src/components/application_manager/src/commands/mobile/system_request.cc

```diff
@@ -103,39 +103,46 @@ class QueryAppsDataValidator {
   }
 
   bool ValidateAppDataAndOsAndLanguagesData() const {
-    const smart_objects::SmartArray* objects_array =
-        data_[json::response].asArray();
+    smart_objects::SmartArray* objects_array = data_[json::response].asArray();
+
     if (!objects_array) {
       LOG4CXX_WARN(logger_,
                    kQueryAppsValidationFailedPrefix
                        << "QueryApps response is not array.");
       return false;
     }
-    const std::size_t arr_size(objects_array->size());
+
     SynonymsMap synonyms_map;
-    for (std::size_t idx = 0; idx < arr_size; ++idx) {
-      const smart_objects::SmartObject& app_data = (*objects_array)[idx];
+    bool has_response_valid_application = false;
+
+    smart_objects::SmartArray::iterator applications_iterator =
+        objects_array->begin();
+
+    for (; applications_iterator != objects_array->end();
+         ++applications_iterator) {
+      const smart_objects::SmartObject& app_data = *applications_iterator;
 
       if (!app_data.isValid()) {
         LOG4CXX_WARN(logger_,
                      kQueryAppsValidationFailedPrefix
                          << "Wrong application data in json file.");
         return false;
       }
-      std::set<std::string> app_ids_set;
-      if (!ValidateAppIdAndAppName(app_data, app_ids_set)) {
+
+      if (!CheckMandatoryParametersPresent(app_data)) {
+        LOG4CXX_WARN(logger_,
+                     "Application hasn`t some of mandatory parameters. "
```

Comment:
@VProdanov , both ways grammatically right, so, I think it's doesn't matter there is.